lib/jsdom/level2/style.js

```diff
@@ -173,6 +173,13 @@ core.HTMLStyleElement._init = function() {
 
     evaluateStylesheet.call(this, content, this.sheet, this._ownerDocument.URL);
   });
+
+  this.addEventListener('DOMNodeRemovedFromDocument', function() {
+    var index = this.ownerDocument.styleSheets.indexOf(this.sheet);
+    if(index > -1) {
+      this.ownerDocument.styleSheets = this.ownerDocument.styleSheets.splice(index, 1);
```

Comment:
You should just do `this.ownerDocument.styleSheets.splice(index, 1)`, without the reassignment.